### PR TITLE
MODINREACH-171: Add REST API for Inn-Reach Recall user

### PR DESCRIPTION
**Pull request by andrew-khodyka**
## Purpose
Add REST API for Inn-Reach Recall User
https://issues.folio.org/browse/MODINREACH-171

## Approach

- Add Inn-Reach Recall User RestController
- Add Inn-Reach Recall User Service
- Add Inn-Reach Recall User DTO

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/InnReachRecallUserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  public InnReachRecallUserDTO updateInnReachRecallUser(UUID centralServerId, InnReachRecallUserDTO innReachRecallUserDTO) {
+    var centralServer = fetchRecallUser(centralServerId);
+
+    var innReachRecallUser = centralServer.getInnReachRecallUser();
+    innReachRecallUser.setUserId(innReachRecallUserDTO.getUserId());
+
+    return recallUserMapper.toDto(centralServer.getInnReachRecallUser());
+  }
+
+  private CentralServer fetchRecallUser(UUID centralServerId) {
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/InnReachRecallUserServiceImpl.java]: `fetchCentralServer` would suit better

File: src/test/java/org/folio/innreach/controller/InnReachRecallUserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  @Test
+  @Sql(scripts = {
+    "classpath:db/inn-reach-recall-user/pre-populate-inn-reach-recall-user.sql",
+    "classpath:db/central-server/pre-populate-central-server-with-recall-user.sql"
+  })
+  void returnErrorMessage_when_centralServerRecallUserDoesNotExist() {
+    var responseEntity = testRestTemplate.getForEntity("/inn-reach/central-servers/{id}/inn-reach-recall-user",
+        InnReachRecallUserDTO.class, UUID.fromString("edab6baf-c697-42b2-89bb-1bbb8759b0d3"));
```
Comment (oliinyko) [inline src/test/java/org/folio/innreach/controller/InnReachRecallUserControllerTest.java]: the same request path and uuid are used among multiple tests, it's better to set them as constants

File: src/main/java/org/folio/innreach/repository/CentralServerRepository.java [hunk 3]
```diff
@@ -37,4 +37,7 @@
   @Query(name = CentralServer.GET_ID_BY_CENTRAL_CODE_QUERY_NAME)
   Optional<UUID> getIdByCentralCode(String centralCode);
 
+  @Query(name = CentralServer.FETCH_RECALL_USER_BY_ID_QUERY_NAME)
+  Optional<CentralServer> fetchRecallUser(UUID id);
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/repository/CentralServerRepository.java]: it's more like `fetchOneWithRecallUser`

**Review by oliinyko — APPROVED** [review]

File: src/main/java/org/folio/innreach/domain/service/InnReachRecallUserServiceImpl.java (hunk 1 above)
Comment (dmtkachenko) [reply src/main/java/org/folio/innreach/domain/service/InnReachRecallUserServiceImpl.java]: .. or return innReachRecallUser not CentralServer

**Comment by dmtkachenko** [comment]
ModuleDescriptor-template.json changes are missing

**Review by dmtkachenko — CHANGES_REQUESTED** [review]
ModuleDescriptor-template.json changes are missing

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -300,6 +300,21 @@
           "pathPattern": "/inn-reach/central-servers/{centralServerId}/central-patron-type-mappings",
           "permissionsRequired": ["inn-reach.central-patron-type-mappings.collection.put"]
         },
+        {
+          "methods": ["GET"],
+          "pathPattern": "/inn-reach/central-servers/{centralServerId}/inn-reach-recall-user",
+          "permissionsRequired": ["inn-reach.inn-reach-recall-user.item.get"]
```
Comment (dmtkachenko) [inline descriptors/ModuleDescriptor-template.json]: Please use permission names from the JIRA

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**